src/Chunker.php

```diff
@@ -53,92 +39,119 @@ protected function schedule_chunk(array|Iterator $chunkData): void
-
-                    // Remove the processed record from the buffer
-                    $buffer = substr($buffer, $pos + strlen(self::SERIALIZED_DELIMITER));
-
-                    // Deserialize and yield the record if it's not empty
-                    if (!empty(trim($record))) {
-                        yield unserialize(trim($record));
-                    }
-                }
-            }
-
-            // Process any remaining data in the buffer after reading the entire file
-            if (!empty(trim($buffer))) {
-                yield unserialize(trim($buffer));
+        // set the new status of the chunk
+        $chunk = new Chunk( $chunk_id );
+        $chunk->update( array(
+            'status' => 'running'
+        ) );
+
+        // fetch the data
+        $data = $chunk->get_data();
+
+        // Convert array to Generator
+        $generator = (function () use ($data) {
+            foreach ($data as $key => $value) {
+                yield $key => $value;
             }
         })();
 
-        $this->process_chunk_data($formattedDataGenerator);
+        $this->process_chunk_data($generator);
+    }
 
-        fclose($file);
+    /**
+     * Handles the actual data processing. Should be implemented in the class lowest in hierarchy.
+     *
+     * @param Generator<mixed> $chunkData The generator containing chunk data to process
+     * @return void
+     */
+    abstract function process_chunk_data(Generator $chunkData): void;
 
-        // Remove chunk file after sync
-        $unlink_result = unlink($chunk_file_path);
-        if ( $unlink_result === false ) {
-            throw new Exception("File '$chunk_file_path' could not be deleted!");
-        }
+    /**
+     * Schedules the cleanup job if not already scheduled.
+     * Creates a daily cron job at midnight to clean up old chunk data.
+     *
+     * @return void
+     */
+    public function schedule_chunk_cleanup(): void
+    {
+        if ( as_has_scheduled_action( 'ASP/Chunks/Cleanup' ) ) {
+			return;
+		}
+
+        // schedule the cleanup midnight every day
+		as_schedule_cron_action(
+			time(),
+			'0 0 * * *', 'ASP/Chunks/Cleanup'
+		);
     }
 
     /**
-     * Handles the actual data processing. Should be implemented in the class lowest in hierarchy
+     * Cleans the chunk data table from data with following properties:
+     * - older than 2 days (start)
+     * - status must be finished
      *
-     * @param Generator $chunkData
      * @return void
      */
-    abstract function process_chunk_data(Generator $chunkData): void;
+    public function cleanup_chunk_data(): void
+    {
+        /**
+         * Filters the number of days to keep chunk data.
+         *
+         * @param string $interval The interval string (e.g., '-14 days').
+         */
+        $days_interval = apply_filters( 'asp/chunks/cleanup/days', '-14 days' );
+        $cleanup_timestamp = (float) strtotime( $days_interval );
```

Comment:
Not the best implementation since it can introduce bugs by passing positive values. Only allow filtering the integer. As discussed seconds is best. WordPress offers Constants so devs do not need to calculate